### PR TITLE
Storybook: Create a doc for Line with a Table of all args.

### PR DESCRIPTION
**Pull request by nikolasrieble**
## Description
Migrate existing API documentation to Storybook for the Line component.

It turns out that there are WAY more properties than those documented. 
The table autogenerates rows for each existing property. For starters I suggest to not use props in Other and Internal.

I would specifically like feedback on the changes to the README, and the categories in the table. Do these make sense? Should there be less? 

## Related Issue
https://github.com/recharts/recharts/issues/3403

## Screenshots (if appropriate):
Screenshot 2023-02-28 at 22 57 19

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: storybook/stories/API/cartesian/Line.stories.tsx [hunk 1]
```diff
@@ -8,6 +8,59 @@ import { coordinateWithNullY, coordinateData, coordinateWithValueData } from '..
 export default {
   component: Line,
   argTypes: {
+    dataKey: {
```
Comment (nikolasrieble) [inline storybook/stories/API/cartesian/Line.stories.tsx:11]: WIP, migrating all documentation from https://recharts.org/en-US/api/Line

**Comment by nikolasrieble** [comment]
FYI @Yilun-Sun

File: storybook/README.md [hunk 2]
```diff
@@ -50,6 +57,8 @@ Additionally, please follow the best practices of Storybook:
 - https://storybook.js.org/docs/7.0/react/essentials/controls
 - https://storybook.js.org/docs/7.0/react/essentials/interactions
 
+When writing a story, best
```
Comment (ckifer) [inline storybook/README.md:60]: nit: I think you meant to continue here

File: storybook/stories/API/cartesian/Line.stories.tsx [hunk 3]
```diff
@@ -5,22 +5,358 @@ import { expect } from '@storybook/jest';
 import { Surface, Line, ResponsiveContainer, ComposedChart } from '../../../../src';
 import { coordinateWithNullY, coordinateData } from '../../data';
 
+// TODO: We might want to move some of these to a shared file.
+// We have a lot of duplication between the stories. Event Handlers are exposed in multiple components.
```
Comment (ckifer) [inline storybook/stories/API/cartesian/Line.stories.tsx:9]: oh wow, by might I think definitely 😅. Deifnitely don't want to repeat this everywhere if we can help it

File: storybook/stories/API/cartesian/Line.stories.tsx [hunk 4]
```diff
@@ -5,22 +5,358 @@ import { expect } from '@storybook/jest';
...
+        type: {
+          summary: 'boolean',
+        },
+        category: 'Data',
       },
+      defaultValue: false,
     },
-    stroke: {
-      control: { type: 'color' },
+    dangerouslySetInnerHTML: { table: { category: 'Other' } },
```
Comment (ckifer) [inline storybook/stories/API/cartesian/Line.stories.tsx:60]: 😢

File: storybook/stories/API/cartesian/Line.stories.tsx [hunk 5]
```diff
@@ -5,22 +5,358 @@ import { expect } from '@storybook/jest';
...
     },
     fill: {
       control: { type: 'color' },
+      table: { category: 'Style' },
+    },
+    hide: {
+      description: 'Hides the line when true, useful when toggling visibility state via legend',
+      type: { name: 'boolean' },
+      defaultValue: false,
+      table: { category: 'Style' },
```
Comment (ckifer) [inline storybook/stories/API/cartesian/Line.stories.tsx:94]: nit: general comment - should we put all properties that are in the same category adjacent to eachother in this JSON? Might help just for code organization

**Review by ckifer — CHANGES_REQUESTED** [review]
Nice! Just the typo, the rest are nits/later things

File: storybook/stories/API/cartesian/Line.stories.tsx (hunk 5 above)
Comment (nikolasrieble) [reply storybook/stories/API/cartesian/Line.stories.tsx:94]: Good point.

File: storybook/stories/API/cartesian/Line.stories.tsx (hunk 3 above)
Comment (nikolasrieble) [reply storybook/stories/API/cartesian/Line.stories.tsx:9]: Yeah, I am not yet certain on which properties are shared where, once I figure that out the organisation of the props will have to emerge.

Comment (nikolasrieble) [reply storybook/stories/API/cartesian/Line.stories.tsx:9]: Good point, I right away took out three parts - EventHandlers, Animation and Legend for reuse.

File: storybook/stories/API/cartesian/Line.stories.tsx (hunk 5 above)
Comment (nikolasrieble) [reply storybook/stories/API/cartesian/Line.stories.tsx:94]: I improved the organisation. For some props I am not certain whether my suggested category makes sense. We will most likely iterate here as we add more ArgsTables.

File: storybook/stories/API/cartesian/Line.stories.tsx [hunk 6]
```diff
@@ -4,42 +4,218 @@ import { within } from '@storybook/testing-library';
...
+import { legendType } from '../props/Legend';
 
-export default {
-  component: Line,
-  argTypes: {
-    connectNulls: {
-      control: {
-        type: 'boolean',
+// TODO: We might want to move some of these to a shared file.
+// We have a lot of duplication between the stories. Event Handlers are exposed in multiple components.
```
Comment (nikolasrieble) [inline storybook/stories/API/cartesian/Line.stories.tsx:12]:
```suggestion
```

**Comment by nikolasrieble** [comment]
One thing I did not yet figure out but would like to improve is the order of the groups in the table 🤔

File: storybook/stories/API/props/EventHandlers.ts [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+export const onAnimationEndCapture = { table: { category: 'EventHandlers' } };
+export const onAnimationIteration = { table: { category: 'EventHandlers' } };
+export const onAnimationIterationCapture = { table: { category: 'EventHandlers' } };
+export const onAnimationStart = {
+  table: {
+    description: 'The customized event handler of animation start',
+    category: 'EventHandlers',
+  },
+};
+export const onAnimationStartCapture = { table: { category: 'EventHandlers' } };
```
Comment (ckifer) [inline storybook/stories/API/props/EventHandlers.ts:23]: idk if everyone of these need exported individually but eh, looks good

**Review by ckifer — APPROVED** [review]
🚀

File: storybook/stories/API/props/EventHandlers.ts (hunk 7 above)
Comment (nikolasrieble) [reply storybook/stories/API/props/EventHandlers.ts:23]: Probably not, but am not certain whether all components support all EventHandler? 🤔

**Pull request merged**